share/functions/ls.fish

```diff
@@ -17,18 +17,27 @@ function __fish_set_lscolors --description 'Set $LS_COLORS if possible'
     end
 end
 
+function __fish_ls_command --description 'Use colorls if it is installed'
```

Comment:
I probably should have been clearer: I meant either do this check in the actual `ls` function, or define an entirely different `ls` function that uses `colorls` if available. But now I see that you'd still want to call `colorls -F` (I assumed it did *just* color), so it's probably simplest to just inline this check in the actual `ls` function.

This indirection is just as cheesy as having a global variable.

Reply:
Got it. I was thinking it would be preferable to use the color __fish_ls_color_opt code to test ``colorls`` for supported options before using them, in case some other version of ``colorls`` exists in the world and behaves differently, but perhaps I was over engineering it...if such a ``colorls`` exists, it could very well have different options behaviors that would need to be dealt with. Better to just wait for those to come up than to guess.

Anyhoo, I'll take another shot it. Thanks again for the feedback. :)